### PR TITLE
Feature: Staking

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine-interface/src/model/serialize.rs [hunk 1]
```diff
@@ -29,6 +31,29 @@ pub enum NativeInvocation {
     TransactionRuntime(TransactionRuntimeInvocation),
 }
 
+impl NativeInvocation {
+    pub fn replace_ids(
```
Comment (iamyulong) [inline radix-engine-interface/src/model/serialize.rs:35]: This is interesting!

File: radix-engine/src/ledger/bootstrap.rs [hunk 2]
```diff
@@ -197,18 +180,49 @@ pub fn create_genesis(
...
+        for (key, amount) in validator_set {
+            let bucket = id_allocator.new_bucket_id().unwrap();
+            instructions.push(
+                BasicInstruction::TakeFromWorktopByAmount {
+                    resource_address: RADIX_TOKEN,
+                    amount,
+                }
+                .into(),
+            );
+            validators.insert(key, Bucket(bucket.0));
```
Comment (iamyulong) [inline radix-engine/src/ledger/bootstrap.rs:193]: `ManifestBucket`?

File: radix-engine/src/model/epoch_manager/executables.rs [hunk 3]
```diff
@@ -64,11 +71,19 @@ impl Executor for EpochManagerCreateInvocation {
...
         };
 
-        let mut validator_set = BTreeSet::new();
+        let mut validator_set = BTreeMap::new();
 
-        for key in self.validator_set {
-            let address = EpochManager::create_validator(global_node_id.into(), key, api)?;
-            validator_set.insert(Validator { address, key });
+        for (key, bucket) in self.validator_set {
+            let stake = Decimal::one();
```
Comment (iamyulong) [inline radix-engine/src/model/epoch_manager/executables.rs:77]: Is this  `stake_units` or `staked_xrd_amount`? It doesn't seem to be set according to the bucket.

**Review by iamyulong — COMMENTED** [review]
LGTM with some questions.

File: radix-engine/src/model/epoch_manager/executables.rs (hunk 3 above)
Comment (talekhinezh) [reply radix-engine/src/model/epoch_manager/executables.rs:77]: ooo yup good catch

**Review by iamyulong — APPROVED** [review]
Approved. There is weirdness regarding bucket replacements in native invocation, but will be resolved as part of the unified invocation paths.

**Pull request merged**